tumblr/k8s-sidecar-injector
# Feature/inject volume mounts

First off: Thank you for open-sourcing this amazing project :smiley: 

# What and why?

Feature to add volume mounts to all existing and injected containers by adding top level `volumeMounts` scope. This is useful when an injected sidecar container is supposed to share files with the existing containers. My use case for this was a Kerberos sidecar container that provides Kerberos tickets to the other containers in the pod.

# Testing Steps

Tested within microk8s and minikube on Ubuntu 18.04 by adding the aforementioned fields to the example in `examples/kubernetes` and then checking if all the containers share the same configmap mounted.
Unfortunately there are no unittests yet, as the current setup (load InjectionConfig and expected IC via struct from JSON annotations) can't appropriately be applied to this case (because we need to check that each container has this one volumeMount afterwards and not count the global number).
If we can clarify a procedure, I'd like to add the tests later :+1: 

UPDATE (2019-01-28): I added unit tests, which require a bit more complexity, because `volumeMounts` are top-level (global scope) objects in the injection-config, but a second-level (containers scope) objects in the resulting config (after injection), which didn't really work with the current 1:1 mapping.

EDIT: seems like I maybe misunderstood how your unittests work :thinking: Anyways, I'll just wait for your feedback :sweat_smile:
